src/components/autosizeInput/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, ChangeEvent, FocusEvent, KeyboardEvent } from 'react';
+import styled, { css } from 'styled-components';
+import useAutosizeInput from '../../hooks/useAutosizeInput';
+export interface AutosizeInputProps {
+    type: 'text';
+    maxLength: number;
+    minWidth: number;
+    className?: string;
+    placeholder?: string;
+    label?: string;
+    color?: string;
+    disabled?: boolean;
+    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+    onBlur?: (e: FocusEvent<HTMLInputElement>) => void;
+    onKeyDown?: (e: KeyboardEvent<HTMLInputElement>) => void;
+}
+const BaseInput = (props: AutosizeInputProps): React.ReactElement<AutosizeInputProps> => {
+    const {
+        className,
+        placeholder,
+        minWidth,
+        onChange,
+        ...other
+    } = props;
+    const { AutosizeInputState, onChangeValue, onChangeWidth } = useAutosizeInput();
+    let sizerEle: HTMLSpanElement | null;
+    let placeHolderSizer: HTMLSpanElement | null;
+    let newInputWidth: number;
+
+    const resizeInput = (e: ChangeEvent<HTMLInputElement>) => {
+        onChange(e);
+        if (!sizerEle || typeof sizerEle.scrollWidth === 'undefined') {
+            return;
+        }
+        const changeValue = e.target.value;
+        onChangeValue(changeValue);
+    };
+
+    const updateInputWidth = () => {
+        if (!sizerEle) {
+            return;
+        } else if (placeHolderSizer && !AutosizeInputState.value) {
+            newInputWidth = Math.max(sizerEle.scrollWidth, placeHolderSizer.scrollWidth) + 2;
+        } else {
+            newInputWidth = sizerEle.scrollWidth + 2;
+        }
+        if (newInputWidth <= minWidth) {
+            newInputWidth = minWidth;
+        }
+        onChangeWidth(newInputWidth);
+    };
+
+    useEffect(() => updateInputWidth(), [AutosizeInputState.value]);
+
+    const HiddenText = styled.span`
+        position: absolute;
+        overflow: scroll;
+        white-space: pre;
+        letter-spacing: -0.9px;
+        font-weight: 500;
+        font-size: 1.1em;
+        white-space: pre;
+        display: block;
+        visibility: hidden;
+    `;
+    const dynamicStyle = {
+        width: `${AutosizeInputState.inputWidth}px`,
+    };
+
+    return (
+        <div className={className}>
+            <input
+                onChange={(e) => resizeInput(e)}
```

Comment:
props로 넘어온 onChange는 resize하는 역할이 아닌 것 같아서 resizeInput 내부에서 호출하는 것 보다는 따로 분리하는게 좋을 것 같아요!
```suggestion
                onChange={(e) => {
                  onChnage(e);
                  resizeInput(e); // 해당 함수 안에 있는 onChange 제거
                }}
```

Reply:
넵넵